lib/chef/knife/s3_source.rb

```diff
@@ -20,11 +20,11 @@ def bucket_obj
       end
 
       def bucket
-        URI(@url).host
+        URI(@url).path.split("/")[1]
```

Comment:
@eherot , I am a bit confused about how does user get this URL. I referred `Accessing a Bucket` section here http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html, but it too shows URL with `http://` format. Does user have to create this `s3://` path manually that you suggested?